storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorUtils.java

```diff
@@ -0,0 +1,353 @@
+            final Utils.ExitCodeCallable exitCodeCallback, File dir) throws IOException {
+        if (StringUtils.isBlank(user)) {
+            throw new IllegalArgumentException("User cannot be blank when calling workerLauncher.");
+        }
+        String wlinitial = (String) (conf.get(Config.SUPERVISOR_WORKER_LAUNCHER));
+        String stormHome = ConfigUtils.concatIfNotNull(System.getProperty("storm.home"));
+        String wl;
+        if (StringUtils.isNotBlank(wlinitial)) {
+            wl = wlinitial;
+        } else {
+            wl = stormHome + "/bin/worker-launcher";
+        }
+        List<String> commands = new ArrayList<>();
+        commands.add(wl);
+        commands.add(user);
+        commands.addAll(args);
+        LOG.info("Running as user: {} command: {}", user, commands);
+        return Utils.launchProcess(commands, environment, logPreFix, exitCodeCallback, dir);
+    }
+
+    public static int workerLauncherAndWait(Map conf, String user, List<String> args, final Map<String, String> environment, final String logPreFix)
+            throws IOException {
+        int ret = 0;
+        Process process = workerLauncher(conf, user, args, environment, logPreFix, null, null);
+        if (StringUtils.isNotBlank(logPreFix))
+            Utils.readAndLogStream(logPreFix, process.getInputStream());
+        try {
+            process.waitFor();
+        } catch (InterruptedException e) {
+            LOG.info("{} interrupted.", logPreFix);
+        }
+        ret = process.exitValue();
+        return ret;
+    }
+
+    public static void setupStormCodeDir(Map conf, Map stormConf, String dir) throws IOException {
+        if (Utils.getBoolean(conf.get(Config.SUPERVISOR_RUN_WORKER_AS_USER), false)) {
+            String logPrefix = "setup conf for " + dir;
+            List<String> commands = new ArrayList<>();
+            commands.add("code-dir");
+            commands.add(dir);
+            workerLauncherAndWait(conf, (String) (stormConf.get(Config.TOPOLOGY_SUBMITTER_USER)), commands, null, logPrefix);
+        }
+    }
+
+    public static void rmrAsUser(Map conf, String id, String path) throws IOException {
+        String user = Utils.getFileOwner(path);
+        String logPreFix = "rmr " + id;
+        List<String> commands = new ArrayList<>();
+        commands.add("rmr");
+        commands.add(path);
+        SupervisorUtils.workerLauncherAndWait(conf, user, commands, null, logPreFix);
+        if (Utils.checkFileExists(path)) {
+            throw new RuntimeException(path + " was not deleted.");
+        }
+    }
+
+    /**
+     * Given the blob information returns the value of the uncompress field, handling it either being a string or a boolean value, or if it's not specified then
+     * returns false
+     * 
+     * @param blobInfo
+     * @return
+     */
+    public static Boolean shouldUncompressBlob(Map<String, Object> blobInfo) {
+        return new Boolean((String) blobInfo.get("uncompress"));
```

Comment:
It only handles string values and not boolean values. Correct definition would be
` return new Boolean(String.valueOf(blobInfo.get("uncompress")))`